docs/spec/charts.js

```diff
@@ -1,4 +1,12 @@
-/* global createCollaborationChart, createHistoryChart, createList, createTable, createSpinner */
+/* global
+    aggregateTimeData,
+    createCollaborationChart,
+    createHistoryChart,
+    createList,
+    createTable,
+    createSpinner,
+    d3,
```

Comment:
Globals can be defined in a variety of ways in eslint. Other than inlining the list into particular .js files (like we do here), you can also specify them via eslint config files, similarly to how [we do in the browser-context specific `.eslintrc.json` file](https://github.com/Autodesk/hubble/blob/master/docs/assets/js/.eslintrc.json#L5-L13) (both `env` and `globals` config properties, in their own different ways, define what globals are expected to be available). I think we should move any globals defined by external libraries that our scripts depend on, like d3, jquery and moment.js, for example, to the configuration files, rather than as per-js-file running lists. In this way, if our webapp frontend ends up adding a new .js file, we won't have to duplicate the inline comment allowing for d3 globals.

Reply:
Thanks, that’s a good pointer. I wasn’t happy with inlining these globals either, and I second that it would be best to configure this in the `.eslintrc.json`.

I’ll return to this pull request in a couple of days. I’ll have to rebase these changes on top of the other new stuff, and then I’ll also adjust the global directives.

Reply:
@filmaj: I tried to change the [`globals` declaration for `d3`](https://github.com/Autodesk/hubble/blob/master/docs/assets/js/.eslintrc.json#L11) to `true` without success. I’d say let’s tidy up the global declaration in a separate pull request 🙂.

Reply:
Noted, I can take that on.

Reply:
That would be nice, thanks 😄!

Reply:
@pluehne I added another commit to my fork's branch of this PR that should fix that, feel free to pull in to this PR: https://github.com/filmaj/hubble/tree/patrick/improved-aggregation

Reply:
@filmaj: Thanks a lot! What I didn’t understand was that there are multiple `.eslintrc.json` files with different purposes—one for linting the assets themselves and one for linting the unit tests.

I squashed your changes into my commit above (to not have one commit do it improperly and another one fixing it right after), and I hope that this is fine with you 🙂.

Reply:
@pluehne yep perfectly fine 👍